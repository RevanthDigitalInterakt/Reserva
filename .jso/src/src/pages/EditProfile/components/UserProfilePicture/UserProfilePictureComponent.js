  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _asyncToGenerator2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _slicedToArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[2]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[3]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function UserProfilePictureComponent(_ref) {
    var file = _ref.file,
      toogleModalChangeFile = _ref.toogleModalChangeFile,
      userEmail = _ref.userEmail;
    var _useState = (0, _react.useState)(),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      currentPath = _useState2[0],
      setCurrentPath = _useState2[1];
    var navigation = (0, _$$_REQUIRE(_dependencyMap[4]).useNavigation)();
    var _FirebaseService = new (_$$_REQUIRE(_dependencyMap[5]).FirebaseService)(),
      getUrlFS = _FirebaseService.getUrlFS;
    var handleNavigationEditPassword = (0, _react.useCallback)(function () {
      navigation.navigate('EditPassword', {
        email: userEmail
      });
    }, [userEmail]);
    var handleSetCurrentPathImage = (0, _react.useCallback)(/*#__PURE__*/function () {
      var _ref2 = (0, _asyncToGenerator2.default)(function* (uri) {
        if (uri) {
          if (!uri.includes('https://') && !uri.includes('file://')) {
            setCurrentPath(yield getUrlFS(uri));
            return;
          }
          setCurrentPath(uri);
          return;
        }
        setCurrentPath(undefined);
      });
      return function (_x) {
        return _ref2.apply(this, arguments);
      };
    }(), []);
    (0, _react.useEffect)(function () {
      (0, _asyncToGenerator2.default)(function* () {
        yield handleSetCurrentPathImage(file.uri);
      })();
    }, [file]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsxs)(_$$_REQUIRE(_dependencyMap[7]).Box, {
      alignItems: "center",
      children: [!!currentPath && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsx)(_$$_REQUIRE(_dependencyMap[8]).Avatar, {
        imageSource: {
          uri: currentPath
        },
        onPress: toogleModalChangeFile,
        buttonEdit: true,
        imageStyle: {
          width: 60,
          height: 60,
          borderRadius: 60
        }
      }), !currentPath && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsx)(_$$_REQUIRE(_dependencyMap[8]).Avatar, {
        onPress: toogleModalChangeFile,
        imageStyle: {
          width: 60,
          height: 60,
          borderRadius: 60
        },
        buttonEdit: true
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsx)(_$$_REQUIRE(_dependencyMap[7]).Box, {
        justifyContent: "flex-start",
        alignItems: "flex-start",
        marginTop: "micro",
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsx)(_$$_REQUIRE(_dependencyMap[9]).Button, {
          inline: true,
          onPress: handleNavigationEditPassword,
          title: "Alterar senha",
          testID: "com.usereserva:id/userprofilepicture_button_edit_password",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[6]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Typography, {
            style: {
              textDecorationLine: 'underline'
            },
            fontSize: 13,
            fontFamily: "nunitoRegular",
            children: "Alterar senha"
          })
        })
      })]
    });
  }
  var _default = exports.default = UserProfilePictureComponent;
