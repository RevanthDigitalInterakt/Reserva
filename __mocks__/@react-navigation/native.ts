const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
