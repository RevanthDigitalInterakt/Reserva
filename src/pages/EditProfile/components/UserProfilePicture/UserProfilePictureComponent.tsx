import {
  Avatar, Box, Button, Typography,
} from '@usereservaapp/reserva-ui';
import React, { useCallback, useEffect, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import type { IFile } from '../ModalChangeFile/ChangeFileModal';
import { FirebaseService } from '../../../../shared/services/FirebaseService';

interface IUserProfilePictureComponentProps {
  file: IFile;
  toogleModalChangeFile: () => void,
  userEmail: string
}

function UserProfilePictureComponent({
  file,
  toogleModalChangeFile, userEmail,
}: IUserProfilePictureComponentProps): JSX.Element {
  const [currentPath, setCurrentPath] = useState<string>();
  const navigation = useNavigation();
  const { getUrlFS } = new FirebaseService();

  const handleNavigationEditPassword = useCallback(() => {
    navigation.navigate('EditPassword', {
      email: userEmail,
    });
  }, [userEmail]);

  const handleSetCurrentPathImage = useCallback(async (uri?: string) => {
    if (uri) {
      if (!uri.includes('https://') && !uri.includes('file://')) {
        setCurrentPath(await getUrlFS(uri));
        return;
      }
      setCurrentPath(uri);

      return;
    }

    setCurrentPath(undefined);
  }, []);

  useEffect(() => {
    (async () => {
      await handleSetCurrentPathImage(file.uri);
    })();
  }, [file]);

  return (
    <Box alignItems="center">
      {!!currentPath && (
      <Avatar
        imageSource={{ uri: currentPath }}
        onPress={toogleModalChangeFile}
        buttonEdit
        imageStyle={{ width: 60, height: 60 }}
      />
      )}

      {!(currentPath) && (
      <Avatar
        onPress={toogleModalChangeFile}
        imageStyle={{ width: 60, height: 60 }}
        buttonEdit
      />
      )}

      <Box
        justifyContent="flex-start"
        alignItems="flex-start"
        marginTop="micro"
      >
        <Button
          inline
          onPress={handleNavigationEditPassword}
          title="Alterar senha"
          testID="userprofilepicture_button_edit_password"
        >
          <Typography
            style={{ textDecorationLine: 'underline' }}
            fontSize={13}
            fontFamily="nunitoRegular"
          >
            Alterar senha
          </Typography>
        </Button>
      </Box>
    </Box>
  );
}

export default UserProfilePictureComponent;
