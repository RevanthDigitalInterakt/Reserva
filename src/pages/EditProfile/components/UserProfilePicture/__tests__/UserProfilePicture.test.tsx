import { ThemeProvider } from 'styled-components/native';
import { theme } from '@usereservaapp/reserva-ui';
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import UserProfilePictureComponent from '../UserProfilePictureComponent';

const mockProps = {
  file: {
    uri: undefined,
    type: '',
    name: '',
    initialFilePath: undefined,
  },
  toogleModalChangeFile: jest.fn(),
  userEmail: 'user@gmail.com.br',
};

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <UserProfilePictureComponent {...mockProps} />
  </ThemeProvider>
);

describe('Submiting Content', () => {
  beforeEach(async () => {
    render(TestingComponent);
  });

  it('renders without error and match snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('check button password is visible', () => {
    expect(screen.getByTestId('userprofilepicture_button_edit_password')).toBeVisible();
  });

  // @TODO adicianar teste para verificar se o avatar esta visivel ou nao
});
