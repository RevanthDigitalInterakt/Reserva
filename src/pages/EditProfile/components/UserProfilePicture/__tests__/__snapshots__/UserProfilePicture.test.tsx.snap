// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submiting Content renders without error and match snapshot 1`] = `
<View
  alignItems="center"
  style={
    Array [
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    alignItems="center"
    bg="neutroFrio1"
    borderRadius="infinity"
    height={60}
    justifyContent="center"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#D8D9DA",
          "borderRadius": 999999,
          "height": 60,
          "justifyContent": "center",
          "width": 60,
        },
      ]
    }
    width={60}
  >
    <View
      alignItems="center"
      bg="white"
      borderRadius="infinity"
      height={60}
      justifyContent="center"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderRadius": 999999,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          },
        ]
      }
      width={60}
    >
      <RNSVGSvgView
        bbHeight={60}
        bbWidth={60}
        color="neutroFrio1"
        focusable={false}
        size={60}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Array [
              Object {
                "color": "#D8D9DA",
                "height": 60,
                "width": 60,
              },
            ],
            Object {
              "flex": 0,
              "height": 60,
              "width": 60,
            },
          ]
        }
      >
        <RNSVGGroup>
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            fill="#D8D9DA"
            focusable={false}
            height="100%"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Array [
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                ],
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={512}
            vbWidth={496}
            width="100%"
          >
            <RNSVGGroup
              fill={4292401626}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      height={25}
      left={44}
      position="absolute"
      right={0}
      style={
        Array [
          Object {
            "height": 25,
            "left": 44,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 25,
          },
        ]
      }
      top={0}
      width={25}
    >
      <View
        alignItems="center"
        height={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 25,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8A8C8E",
              "borderRadius": 999999,
              "height": 25,
              "justifyContent": "center",
              "opacity": 1,
              "width": 25,
            }
          }
        >
          <RNSVGSvgView
            bbHeight={17.5}
            bbWidth={17.5}
            color={4294967295}
            focusable={false}
            size={17.5}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Array [
                  Object {
                    "color": "#FFF",
                    "height": 17.5,
                    "width": 17.5,
                  },
                ],
                Object {
                  "flex": 0,
                  "height": 17,
                  "width": 17,
                },
              ]
            }
            tintColor={4294967295}
          >
            <RNSVGGroup>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                fill="#FFF"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                    ],
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width="100%"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
  <View
    alignItems="flex-start"
    justifyContent="flex-start"
    marginTop="micro"
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "justifyContent": "flex-start",
          "marginTop": 12,
        },
      ]
    }
  >
    <View
      alignItems="stretch"
      style={
        Array [
          Object {
            "alignItems": "stretch",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.usereserva:id/userprofilepicture_button_edit_password"
      >
        <Text
          fontFamily="nunitoRegular"
          fontSize={13}
          style={
            Array [
              Object {
                "fontFamily": "Nunito-Regular",
                "fontSize": 13,
              },
              Object {
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Alterar senha
        </Text>
      </View>
    </View>
  </View>
</View>
`;
